### PR TITLE
Fix publishing of images

### PR DESCRIPTION
**Pull request by DavidWatkins**
The current publisher for RGB and depth images is slow. I have made the following improvements
- Added publishing the correct scoped frame_id for images
- Removed image publishing from spot_ros2.py and created a separate node to publish rgb, depth, or depth_registered
- Added arguments to publish depth_registered or depth aligned with the rgb frame

### CODE REVIEW

File: spot_driver/spot_driver/publish_cameras.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+import bosdyn.client.util
+from bosdyn.api import image_pb2
+from bosdyn.client.image import ImageClient, build_image_request
+from google.protobuf.timestamp_pb2 import Timestamp
+from builtin_interfaces.msg import Time, Duration
+from spot_driver.ros_helpers import get_from_env_and_fall_back_to_param
+import rclpy.node
+
+
+ROTATION_ANGLE = {
```
Comment (jbarry-bdai) [inline spot_driver/spot_driver/publish_cameras.py:19]: This is in degrees?  That's a little unusual :)

File: spot_driver/spot_driver/spot_ros2.py [hunk 2]
```diff
@@ -226,73 +223,6 @@ def WorldObjectsCB(self, results):
...
-            self.right_image_pub.publish(image_msg1)
-            self.right_image_info_pub.publish(camera_info_msg1)
-            image_msg2, camera_info_msg2 = getImageMsg(data[2], self.spot_wrapper)
-            self.left_depth_pub.publish(image_msg2)
-            self.left_depth_info_pub.publish(camera_info_msg2)
-            image_msg3, camera_info_msg3 = getImageMsg(data[3], self.spot_wrapper)
-            self.right_depth_pub.publish(image_msg3)
-            self.right_depth_info_pub.publish(camera_info_msg3)
-
-            self.populate_camera_static_transforms(data[0])
```
Comment (jbarry-bdai) [inline spot_driver/spot_driver/spot_ros2.py:274]: how are these transforms getting published now?

**Review by jbarry-bdai — COMMENTED** [review]
Before committing this I'd like to understand better if we can put it all in one place.  I don't love having so much duplicated code and authentication etc...

File: spot_driver/launch/point_cloud_xyz.launch.py [hunk 3]
```diff
@@ -37,22 +37,21 @@
 
 import launch_ros.actions
 import launch_ros.descriptions
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration, PathJoinSubstitution
```
Comment (jbarry-bdai) [inline spot_driver/launch/point_cloud_xyz.launch.py:41]: Have we tested this file?  (I haven't used it in its previous or current state.)

File: spot_driver/spot_driver/ros_helpers.py [hunk 4]
```diff
@@ -127,20 +128,21 @@ def createDefaulCameraInfo():
...
         data: Image proto
-        spot_wrapper: A SpotWrapper object
     Returns:
         (tuple):
             * Image: message of the image captured
             * CameraInfo: message to define the state and config of the camera that took the image
     """
     image_msg = Image()
-    local_time = spot_wrapper.robotToLocalTime(data.shot.acquisition_time)
+    local_time = spot_wrapper.robot_to_local_time(data.shot.acquisition_time)
```
Comment (jbarry-bdai) [inline spot_driver/spot_driver/ros_helpers.py:143]: While stylistically correct, this seems like maybe an unnecessary change for this PR?  I'm worried we'll miss a call to it somewhere.

**Review by jbarry-bdai — COMMENTED** [review]
This looks good to me except for changing the robot to local function call - since that's a public function I think at this point we need to leave it the way it is because we don't know who might be using it outside of this.

I'd like to see someone from MASKOR review as well.

File: spot_driver/launch/point_cloud_xyz.launch.py (hunk 3 above)
Comment (DavidWatkins) [reply spot_driver/launch/point_cloud_xyz.launch.py:41]: Yes it still works but it is very slow because the depth publishing is slow.

File: spot_driver/spot_driver/ros_helpers.py (hunk 4 above)
Comment (DavidWatkins) [reply spot_driver/spot_driver/ros_helpers.py:143]: I got all of them in the repository and the code still runs, but if you'd like me to undo it I will.

Comment (jbarry-bdai) [reply spot_driver/spot_driver/ros_helpers.py]: I'm worried about external users like MASKOR.  @METEORITENMAX wdyt?

Comment (DavidWatkins) [reply spot_driver/spot_driver/ros_helpers.py]: I went ahead an undid these changes.

**Review by jbarry-bdai — APPROVED** [review]

**Comment by jbarry-bdai** [comment]
(Kept the branch because we're still pointing at it)

**Comment by skpawar1305** [comment]
also, somehow CPU usage reduced when publishing images was done in a separate node.
I personally used to use it before https://github.com/skpawar1305/spot_ros2/blob/main/spot_driver/spot_driver/get_images.py
n it was always stable
